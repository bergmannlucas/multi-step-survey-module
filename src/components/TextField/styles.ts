import styled, { css } from 'styled-components'

export const Input = styled.input`
  ${({ theme }) => css`
    width: 100%;
    background: transparent;
    border: 0px solid;
    color: ${theme.colors.purple};
    border-bottom: 4px solid ${theme.colors.purple};
    font-size: ${theme.font.sizes.xlarge};
    margin-bottom: ${theme.spacings.small};
  `}
`
