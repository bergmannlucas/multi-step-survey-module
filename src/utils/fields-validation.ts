import Joi from 'joi'

export const schema = Joi.array().items(
  Joi.object({
    name: Joi.string().empty(''),
    email: Joi.string()
      .email({ tlds: { allow: false } })
      .empty('')
  }),
  Joi.object({
    gender: Joi.string().required(),
    age: Joi.number().required()
  }),
  Joi.object({
    book: Joi.string().required(),
    colors: Joi.object({
      black: Joi.boolean(),
      white: Joi.boolean(),
      red: Joi.boolean(),
      green: Joi.boolean(),
      blue: Joi.boolean(),
      yellow: Joi.boolean(),
      purple: Joi.boolean(),
      orange: Joi.boolean(),
      pink: Joi.boolean()
    })
  })
)
