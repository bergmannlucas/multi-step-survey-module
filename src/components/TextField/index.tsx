import { InputHTMLAttributes } from 'react'

import * as S from './styles'

type TextFieldProps = InputHTMLAttributes<HTMLInputElement>

function TextField(props: TextFieldProps) {
  return <S.Input type="text" {...props} />
}

export default TextField
