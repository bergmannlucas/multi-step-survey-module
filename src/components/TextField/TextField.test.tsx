import userEvent from '@testing-library/user-event'
import { render, screen, waitFor } from 'utils/test-utils'

import TextField from '.'

describe('<TextField />', () => {
  it('should render with placeholder', () => {
    const { container } = render(<TextField placeholder="E-mail" />)

    expect(screen.getByPlaceholderText('E-mail')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should change its value when typing', async () => {
    const onInputChange = jest.fn()
    render(<TextField onChange={onInputChange} name="TextField" />)

    const input = screen.getByRole('textbox')
    const text = 'This is my new text'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(onInputChange).toHaveBeenCalledTimes(text.length)
    })
  })
})
