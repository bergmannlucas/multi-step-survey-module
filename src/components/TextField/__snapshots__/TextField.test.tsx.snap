// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> should render with placeholder 1`] = `
.c0 {
  width: 100%;
  background: transparent;
  border: 0px solid;
  color: #9249FF;
  border-bottom: 4px solid #9249FF;
  font-size: 2.0rem;
  margin-bottom: 2.4rem;
}

<input
  class="c0"
  placeholder="E-mail"
  type="text"
/>
`;
